elbepack/initvmaction.py

```diff
@@ -268,7 +404,38 @@ class AttachAction(InitVMAction):
     def __init__(self, node):
         InitVMAction.__init__(self, node)
 
-    def execute(self, _initvmdir, _opt, _args):
+    def _attach_qemu_vm(self, initvmdir):
+        """
+        Attach to QEMU initvm.
+
+        This method is using socat to connect ot the unix-socket of the initvm
+        where the serial console is bound to.
+        """
+        if initvmdir == None:
+            if os.path.isdir('./initvm'):
+                print('Using default initvm directory "./initvm".')
+                initvmdir = './initvm'
+            else:
+                print('No initvm found!')    
+                sys.exit(207)
+
+        # test if socat command is available
+        try:
+            system('which socat > /dev/null 2>&1')
+        except CommandError:
+            print(f'The command "socat" is required.\nPlease install socat: "sudo apt install socat"')
+            sys.exit(208)
+        
+        # connect to socket file if it exists
+        if os.path.exists(os.path.join(initvmdir, 'vm-serial-socket')):
+            system(f'cd {initvmdir} && socat stdin,raw,echo=0,escape=0x11 unix-connect:vm-serial-socket')
```

Comment:
Can this not also use the `socket` module?

Reply:
I have not found a way to get an interactive shell with the socket module (without implementing it).